src/audio/pipeline.c

```diff
@@ -761,6 +762,68 @@ static int pipeline_trigger_on_core(struct pipeline *p, struct comp_dev *host,
 	return ret;
 }
 
+/*
```

Comment:
Can you describe in the commit message what real error this is trying to solve as it looks like it add unnecessary logic.

Reply:
The Xrun recovery doesn't work with today's code, so this is cleanup and refactor to the xrun recover logic/sequence.

Maybe I can add some comments inside source to describe the new Xrun recovery logic/sequence?